pkg/codegen/python/gen.go

```diff
@@ -1474,6 +1475,11 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
 	return files, nil
 }
 
+func PrepareOutDir(outDir string, pkgName string) error {
```

Comment:
I think this should also take a set of files we want to retain and then pass that to `CleanDir`. Consumers can then just pass in the overlays. Same for node.